### PR TITLE
Add initial psql support

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

**Comment by mfussenegger** [comment]
@dobe please review

File: blackbox/docs/configuration.txt [hunk 1]
```diff
@@ -366,7 +366,7 @@ Ports
   | *Runtime:*   ``no``
 
   This defines the TCP port range to which the Crate HTTP service will be bound to.
-  It defaults to ``4200-4300``. Always the first free port in this range is used. If this is set
+  It defaults to ``4200-4249``. Always the first free port in this range is used. If this is set
```
Comment (dobe) [inline blackbox/docs/configuration.txt]: pls merge port PR

File: blackbox/docs/configuration.txt [hunk 2]
```diff
@@ -403,6 +403,15 @@ Ports
   of the node is not directly reachable from outside, e.g. running it
   behind a firewall or inside a Docker container.
 
+.. _psql.port:
+
+**psql.port**
+  | *Runtime:*   ``no``
+
+  This defines the TCP port range to which the Crate Postgres service will be bound to.
+  It defaults to ``5432-5482``. Always the first free port in this range is used. If this is set
```
Comment (dobe) [inline blackbox/docs/configuration.txt]: use also 100 range

File: blackbox/docs/protocols/postgres.txt [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+Crate contains experimental support for the `Postgresql wire protocol v3`_.
+
+By default this protocol is **disabled**. To enable it set ``psql.enabled`` to
+true.
+
+If a node is started with postgres support enabled it will by default bind to
+``0.0.0.0:5432``. See :doc:`Configuration <../configuration>` for options to
+change that.
+
```
Comment (dobe) [inline blackbox/docs/protocols/postgres.txt]: whitespace

File: sql/src/main/java/io/crate/metadata/settings/CrateSettings.java [hunk 4]
```diff
@@ -1258,14 +1258,58 @@ public boolean isRuntime() {
 
     public static final StringSetting UDC_URL = new StringSetting("url", null, true, "https://udc.crate.io", UDC);
 
+    public static final NestedSetting NETWORK = new NestedSetting() {
```
Comment (dobe) [inline sql/src/main/java/io/crate/metadata/settings/CrateSettings.java]: use psql.* like http.*

File: sql/src/main/java/io/crate/operation/projectors/fetch/FetchProjector.java [hunk 5]
```diff
@@ -127,6 +135,7 @@ public FetchProjector(FetchOperation fetchOperation,
     }
 
     private boolean nextStage(Stage from, Stage to) {
+        LOGGER.debug("Changing state from {} to {}", from, to);
```
Comment (dobe) [inline sql/src/main/java/io/crate/operation/projectors/fetch/FetchProjector.java]: trace?

File: blackbox/docs/protocols/postgres.txt (hunk 3 above)
Comment (dobe) [inline blackbox/docs/protocols/postgres.txt]: the space is still too white in this file :)

**Comment by dobe** [comment]
otherwise ack

**Pull request merged**